src/lib.rs

```diff
@@ -213,24 +221,25 @@ impl<T: Send> ThreadLocal<T> {
     }
 
     fn get_inner(&self, thread: Thread) -> Option<&T> {
-        let bucket_ptr =
-            unsafe { self.buckets.get_unchecked(thread.bucket) }.load(Ordering::Acquire);
+        let bucket_ptr =self.get_bucket(thread).load(Ordering::Acquire);
         if bucket_ptr.is_null() {
             return None;
         }
-        unsafe {
-            let entry = &*bucket_ptr.add(thread.index);
-            if entry.present.load(Ordering::Relaxed) {
-                Some(&*(&*entry.value.get()).as_ptr())
-            } else {
-                None
-            }
-        }
+        // SAFETY: Any allocation larger than isize::MAX bytes would fail to
+        // allocate and thus the `bucket` pointer will be null, it thus must
+        // be safe to that offset and create a mutable borrow from it.
+        //
+        // This function has immutable access to the `ThreadLocal` and its contents.
+        // so there should not be concurrent mutable accesses into the same entry.
```

Comment:
The safety comment here should say that `thread.index` is guaranteed to be in bounds for this particular bucket.